client/lib/posts/utils.js

```diff
@@ -217,6 +209,34 @@ var utils = {
 	},
 
 	/**
+	 * Returns the ID of the featured image assigned to the specified post, or
+	 * `undefined` otherwise. A utility function is useful because the format
+	 * of a post varies between the retrieve and update endpoints. When
+	 * retrieving a post, the thumbnail ID is assigned in `post_thumbnail`, but
+	 * in creating a post, the thumbnail ID is assigned to `featured_image`.
+	 *
+	 * @param  {Object} post Post object
+	 * @return {Number}      The featured image ID
+	 */
+	getFeaturedImageId: function( post ) {
+		if ( ! post ) {
+			return;
+		}
+
+		if ( 'featured_image' in post && ! /^https?:\/\//.test( post.featured_image ) ) {
```

Comment:
This line could be another candidate for improvement via Lodash's utility methods.
I think `has( post, 'featured_image' )` and `startsWith( post.featured_image, 'https://' )` or similar might be a little more expressive.

In the JS guidelines we give some mixed messages around using `'value' in object` syntax. We recommend it, but soon-after point out that it has caveats that can lead to "some unexpected scenarios", as in: 
```js
'valueOf' in {}; // true
```
I have a PR open to update these guidelines to focus more lodash's solutions, since it's not yet landed feel free to ignore :)